The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/14797990261/job/41549800201) [(plain)](https://github.com/rust-lang/rust/commit/18d51b81ba6244086a2031da7bd2cfce893cd34f/checks/41549800201/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
info: removing rustup binaries
info: rustup is uninstalled
##[group]Image checksum input
mingw-check-tidy
# We use the ghcr base image because ghcr doesn't have a rate limit
# and the mingw-check-tidy job doesn't cache docker images in CI.
FROM ghcr.io/rust-lang/ubuntu:22.04

ARG DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y --no-install-recommends \
  g++ \
---

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
#
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#12 2.775 Building wheels for collected packages: reuse
#12 2.776   Building wheel for reuse (pyproject.toml): started
#12 2.991   Building wheel for reuse (pyproject.toml): finished with status 'done'
#12 2.992   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132719 sha256=5bb60f62728aaedff7162745ce743c7f2f55069b3e7f82e6a37d70df455797cc
#12 2.992   Stored in directory: /tmp/pip-ephem-wheel-cache-xh6xo_mh/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#12 2.994 Successfully built reuse
#12 2.995 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#12 3.401 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#12 3.402 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 3.940 Collecting virtualenv
#12 3.977   Downloading virtualenv-20.30.0-py3-none-any.whl (4.3 MB)
#12 4.043      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 68.9 MB/s eta 0:00:00
#12 4.099 Collecting platformdirs<5,>=3.9.1
#12 4.102   Downloading platformdirs-4.3.7-py3-none-any.whl (18 kB)
#12 4.138 Collecting filelock<4,>=3.12.2
#12 4.142   Downloading filelock-3.18.0-py3-none-any.whl (16 kB)
#12 4.160 Collecting distlib<1,>=0.3.7
#12 4.164   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#12 4.172      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 81.6 MB/s eta 0:00:00
#12 4.254 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#12 4.440 Successfully installed distlib-0.3.9 filelock-3.18.0 platformdirs-4.3.7 virtualenv-20.30.0
#12 4.441 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 DONE 4.5s

#13 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#13 DONE 0.0s
---
DirectMap4k:      116672 kB
DirectMap2M:     9320448 kB
DirectMap1G:     9437184 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.05s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
[TIMING] core::build_steps::tool::LibcxxVersionTool { target: x86_64-unknown-linux-gnu } -- 0.218
---
         }
-
     }
 
     check_link_info_print_request(sess, &codegen_results.crate_info.crate_types);
fmt: checked 5999 files
Build completed unsuccessfully in 0:01:14
  local time: Fri May  2 15:24:03 UTC 2025
  network time: Fri, 02 May 2025 15:24:04 GMT
##[error]Process completed with exit code 1.
```

</details>

--- a/compiler/rustc_codegen_ssa/src/back/link.rs
+++ b/compiler/rustc_codegen_ssa/src/back/link.rs
@@ -195,9 +195,9 @@ pub fn link_binary(
                 tempfiles_for_stdout_output.push(out_filename);
             }
         }
-
-        check_link_info_print_request(sess, &codegen_results.crate_info.crate_types);
     }
+
+    check_link_info_print_request(sess, &codegen_results.crate_info.crate_types);
 
     // Remove the temporary object file and metadata if we aren't saving temps.
     sess.time("link_binary_remove_temps", || {

